src/components/progress-report/progress-report.repository.ts

```diff
@@ -1,6 +1,95 @@
 import { Injectable } from '@nestjs/common';
+import { node, Query, relation } from 'cypher-query-builder';
+import { Session, UnsecuredDto } from '~/common';
 import { DtoRepository } from '~/core/database';
-import { ProgressReport } from './dto';
+import {
+  ACTIVE,
+  filter,
+  matchPropsAndProjectSensAndScopedRoles,
+  merge,
+  oncePerProject,
+  paginate,
+  sortWith,
+} from '~/core/database/query';
+import { engagementFilters } from '../engagement/engagement.repository';
+import { progressReportSorters } from '../periodic-report/periodic-report.repository';
+import {
+  ProgressReport,
+  ProgressReportFilters,
+  ProgressReportListInput,
+} from './dto';
+import { ProgressReportExtraForPeriodicInterfaceRepository } from './progress-report-extra-for-periodic-interface.repository';
 
 @Injectable()
-export class ProgressReportRepository extends DtoRepository(ProgressReport) {}
+export class ProgressReportRepository extends DtoRepository<
+  typeof ProgressReport,
+  [session: Session]
+>(ProgressReport) {
+  constructor(
+    private readonly extraRepo: ProgressReportExtraForPeriodicInterfaceRepository,
+  ) {
+    super();
+  }
+
+  async list(input: ProgressReportListInput, session: Session) {
+    const result = await this.db
+      .query()
+      .match([
+        node('node', 'ProgressReport'),
+        relation('in', '', 'report'),
+        node('', 'Engagement'),
+        relation('in', '', 'engagement'),
+        node('project', 'Project'),
+      ])
+      .apply(progressReportFilters(input.filter))
+      .apply(
+        this.privileges.forUser(session).filterToReadable({
+          wrapContext: oncePerProject,
+        }),
+      )
+      .apply(sortWith(progressReportSorters, input))
+      .apply(paginate(input, this.hydrate(session)))
+      .first();
+    return result!; // result from paginate() will always have 1 row.
+  }
+
+  protected hydrate(session: Session) {
+    return (query: Query) =>
+      query
+        .match([
+          node('node'),
+          relation('in', '', 'report', ACTIVE),
+          node('parent', 'Engagement'),
```

Comment:
This should target 'LanguageEngagement' specifically due to Internship's generating progress reports unknowingly.

Reply:
Yep will do that for now and address that bug separately 👍